src/btree/bt_ret.c

```diff
@@ -122,7 +122,7 @@ __wt_value_return_buf(
             stop->txnid = unpack.stop_txn;
         }
 
-        return (__wt_page_cell_data_ref(session, page, &unpack, buf));
+        return (buf == NULL ? 0 : __wt_page_cell_data_ref(session, page, &unpack, buf));
```

Comment:
This usage really changes what this function does. You should either:
* Split this into two different functions, one that retrieves timestamps the other that does both.
* Update the name and function level comment to reflect that retrieving the value in the buffer is optional.

Reply:
I'll try to separate them with minimum code duplication.

Reply:
Separated to two functions but they look very similar.